sql/src/test/java/io/crate/jobs/UpsertByIdContextTest.java

```diff
@@ -42,4 +42,57 @@ public void testKill() throws Exception {
         expectedException.expectMessage("Job execution was interrupted by kill");
         future.get();
     }
+
+    @Test
+    public void testStartAfterKill() throws Exception {
+        SymbolBasedShardUpsertRequest request = mock(SymbolBasedShardUpsertRequest.class);
+        SymbolBasedUpsertByIdNode.Item item = mock(SymbolBasedUpsertByIdNode.Item.class);
+        SymbolBasedTransportShardUpsertActionDelegate delegate = mock(SymbolBasedTransportShardUpsertActionDelegate.class);
+
+        SettableFuture<TaskResult> future = SettableFuture.create();
+        UpsertByIdContext context = new UpsertByIdContext(request, item, future, delegate);
+        ContextCallback callback = mock(ContextCallback.class);
+        context.addCallback(callback);
+
+        context.kill();
+        verify(callback, times(1)).onClose();
+        expectedException.expectCause(TestingHelpers.cause(JobKilledException.class));
+        expectedException.expectMessage("Job execution was interrupted by kill");
+        future.get();
+
+        // start does nothing, because the context is already closed
+        context.start();
+        verify(delegate, never()).execute(any(SymbolBasedShardUpsertRequest.class), any(ActionListener.class));
+
+        // close context and verify that the callbacks are not called twice
+        context.close();
+        verify(callback, times(1)).onClose();
+    }
+
+    @Test
+    public void testStartAfterClose() throws Exception {
+        SymbolBasedShardUpsertRequest request = mock(SymbolBasedShardUpsertRequest.class);
```

Comment:
can you extract a method here for the mocking setup to share between the two test methods?